src/shared/ui/link-custom/link-custom.tsx

```diff
@@ -1,43 +1,54 @@
-/* eslint-disable @stylistic/jsx-one-expression-per-line */
-import { AnchorHTMLAttributes, DetailedHTMLProps } from 'react';
+import { AnchorHTMLAttributes, ReactNode } from 'react';
+import { type VariantProps, cva } from 'class-variance-authority';
 import classNames from 'classnames/bind';
 import { Link } from 'react-router-dom';
 
 import styles from './link-custom.module.scss';
 
-type LinkCustomProps = DetailedHTMLProps<
-  AnchorHTMLAttributes<HTMLAnchorElement>,
-  HTMLAnchorElement
-> & {
+export const cx = classNames.bind(styles);
+
+type LinkCustomProps = AnchorHTMLAttributes<HTMLAnchorElement> &
+  VariantProps<typeof linkCustomVariants> &
+  LinkCustomAdditionalProps;
+
+type LinkCustomAdditionalProps = {
   href: string;
-  icon?: JSX.Element;
-  button?: boolean;
-  size?: 'large' | 'medium' | 'small';
-  variant?: 'outlined' | 'colored' | '';
-  rounded?: boolean;
-  className?: string;
+  icon?: ReactNode;
 };
 
+const linkCustomVariants = cva('', {
+  variants: {
+    variant: {
+      roundedSmall: cx('button', 'colored', 'small'),
+      coloredSquare: cx('button', 'colored', 'large'),
```

Comment:
And maybe 'coloredSmall' and 'coloredLarge' for uniformity, but then I'm not sure about shapes.